backend/dataall/modules/datasets/aws/lf_table_client.py

```diff
@@ -22,21 +24,36 @@ def grant_pivot_role_all_table_permissions(self):
         :param table:
```

Comment:
Looking at the `clients` I see your logic of dividing the clients by service (glue, lf) and by dataset-object (dataset, table, storage_location...). At first I thought that it would be easier to have a single service client, but I think this approach by service+ by object makes it easier to follow and allows us to scale more easily. The only duplicated lines would be the "creating the client" lines from 13 to 16 if there are other lf_something_clients, but we save on parameters as having the _table as part of the class, so I would say it pays off. 

The only thing to add, there is some sort of `lf_dataset_client` in the creation of datasets. In `backend/dataall/cdkproxy/assets/gluedatabasecustomresource` This is a custom resource Lambda that uses some LakeFormation methods.